Gems/ROS2/Code/Source/Manipulation/JointsManipulationEditorComponent.cpp

```diff
@@ -50,6 +50,17 @@ namespace ROS2
         incompatible.push_back(AZ_CRC_CE("JointsManipulationService"));
     }
 
+    void JointsManipulationEditorComponent::Activate()
+    {
+        if (m_jointOrderedNames.empty())
```

Comment:
Consider someone running the Editor while this component is in the Level, then changing initial positions.
There would be inconsistency between the two collections. This inconsistency can also occur if someone inputs different joint names.
You can handle changes in editable fields by event attributes (such as OnChangeValidate).

But it is much better to have a single collection, see other comment.